documentation/Using-the-SDK/Advanced/Creating-Your-Own-DataSource.md

```diff
@@ -0,0 +1,204 @@
+# Abstract
+
+This document outlines a scenario where you want to implement a Data Source
+beyond the FileDataSource and DirectoryDataSource provided by the SDK.
+
+# Motivation
+
+For users that have data stored somewhere other than Files or Directories, the
+ability to have their Data Source treated the same as a first class Data Source
+by the SDK is desireable. For example, let's suppose that you are writing many
+Custom Data Sources that get their data from a SQL database, or from a web
+service. Rather than having a file with a URI in it, it would be nice to say
+that your Custom Data Source uses a SqlServerDataSource, or a 
```

Comment:
`Code Syntax` for both `SqlServerDataSource` and `WebServiceDataSource`

also, drop comma